query/workmanager.go

```diff
@@ -0,0 +1,297 @@
+package query
+
+import (
+	"container/heap"
+	"errors"
+	"fmt"
+	"sync"
+)
+
+type batch struct {
+	queries []*Query
+	options *queryOptions
+	errChan chan error
+}
+
+// Config holds the configuration options for a new WorkManager.
+type Config struct {
+	// ConnectedPeers is a function that returns a channel where all
+	// connected peers will be sent. It is assumed that all current peers
+	// will be sent imemdiately, and new peers as they connect.
+	ConnectedPeers func() (<-chan Peer, func(), error)
```

Comment:
`godoc` should define what the function closure return value is.

Reply:
Done: https://github.com/lightninglabs/neutrino/pull/179/commits/38e1bb328ef051ef41a6b5a264cf0294cd4585a6